### ISSUE TITLE
Can I use web-types in Goland without package.json?

### ISSUE THREAD

**Issue by nkev** [JetBrains/web-types#55]
Does the web-types system have a dependency on NPM?

I don't use npm or package.json but I want to use the [HTMX webtype](https://github.com/bigskysoftware/htmx/tree/master/editors/jetbrains) in JetBrains Goland. 

Is this possible?

**Comment by piotrtomiak** [comment]
@nkev so far web-types support in WebStorm works only with `package.json`. But I would love to extend that. Could you please explain a little bit more on how are you using HTMX in your project, specifically, what kind of a build system manages the HTMX dependency?

**Comment by nkev** [comment]
Thanks @piotrtomiak. My build system is very simple and uses Goland's built-in File Watchers to process SCSS and JS bundles:

<img width="900" alt="image" src="https://user-images.githubusercontent.com/5410916/186789324-2faab437-6e27-49db-8254-5fe86fafd0d4.png">

I don't like having hundreds of NPM dependencies, so I manually keep a `lib` folder with subfolders of libraries. Whenever there is a change in my source files or library files the Goland File Watchers run automatically. I just followed the [IntelliJ docs](https://www.jetbrains.com/help/go/transpiling-sass-less-and-scss-to-css.html#ws_sass_less_scss_example_scss) to achieve this and it works well for me.  Here is my library structure (scss folder is hidden here, below lib folder):

<img width="240" alt="image" src="https://user-images.githubusercontent.com/5410916/186791257-df82f5c4-d018-41af-9620-b7fa6d18f5df.png">

**Comment by piotrtomiak** [comment]
Thanks for the explanation. I assume that in lib/htmx for instance there is just the JS file, isn't it?

**Comment by nkev** [comment]
No problem. No, the lib/htmx package is actually quite complex. It even has a web-types file :)

<img width="274" alt="image" src="https://user-images.githubusercontent.com/5410916/188081269-9014d9d8-bc46-4606-9c7d-b25391c8e4e5.png">

**Comment by piotrtomiak** [comment]
Thank you for the example! It looks like any autodetection mechanism wouldn't work for you. You would need a preference page, where you can explicitly specify, which web-types file to enable, correct?

**Comment by nkev** [comment]
Yes, I think an IntelliJ dialog would be more configurable and more explicit than automatically traversing project files to find `**/*.web-types` files

**Comment by anderssv** [comment]
I second this need. :-) Doing a pure HTML/JS page with some custom web-components and getting a lot of "Unknown html tag".

**Comment by brainfoolong** [comment]
Hi there. We also need such a feature, really much. Current "package.json" implementation lacks a lot of power.

For example, a package.json in  subfolder overrides the web-types from it parents. You have a root folder with web-types in package.json, you have a subfolder with no web-types in package.json. Result: In all files edited in this subfolder you will have no web-types support what so ever. Bad for a module based development, where your root contains all base features and components.

In can think of ways to solve this:

- Having separate files with a naming convention that are automatically detected by the IDE, like `.editorconfig` (This would i prefer)
- Having all package.json sum up and not override each other web-types
- Additionally having an option in the scheme, to decide if it is a "root" node, so if this will reset inheritance of parents or not

Our current workaround is: We have the same "web-types" linked in all subfolders that have a package.json. As well as in the root node. Otherwise this wouldn't work. As we have multiple files, it's a bit struggling when a new file arrives...

**Comment by piotrtomiak** [comment]
@brainfoolong - could you please share some example project, so that I can better understand what's the problem and experiment with possible solutions?

**Comment by brainfoolong** [comment]
@piotrtomiak I have prepared a super simple basic project with a root folder and submodules, each of which can (but not must) have a package.json.  Control or changing package.json in submodules is not (always) possible

You have 2 test.html files, on have autocompletion for the expected web-types, one have not.

![image](https://github.com/JetBrains/web-types/assets/1684236/77ccec3e-31a7-4d3d-8c88-e1382bca90e7)

[web-types-test.zip](https://github.com/JetBrains/web-types/files/14919714/web-types-test.zip)

**Comment by piotrtomiak** [comment]
@brainfoolong Thanks! I am looking at some new options to provide Web Types wihtout `package.json` to implement in 2024.2

**Comment by brainfoolong** [comment]
Hey @piotrtomiak - Can we see a preview of some new options in the PhpStorm 2024.2 EAP program? Don't find a trace for that in the release notes.

**Comment by kirach** [comment]
Just wanted to +1 on this, I'm also using htmx and shoelace in my project (which is a super simple app using ktor with server side rendering). I'm including htmx and shoelace in my index.html using `<script ...` tag and I don't use any build tools. The only way to add autocomplete support is to create package.json file with only one line `"web-types": "./web-types.json"`

**Comment by piotrtomiak** [comment]
@brainfoolong - I had in plans to add this support, but I have spent 3 months on implementing new solution for type checking in Angular. It was hell. I am going for vacations for few weeks. I want to add the support for 2024.3 and maybe we could cherrypick it to 2024.2.x.

**Comment by brainfoolong** [comment]
@piotrtomiak Hi, back again :) I'm currently testing 2024.3 EAP. Had you time to implement this thing? Can't see changelog news for anything related. Greetz

**Comment by piotrtomiak** [comment]
@brainfoolong - I was still stuck with this new type evaluation support in WebStorm for TypeScript and Angular. We have planned a lot of improvement for Web Types, but we had to postpone them, unfortunately. I can't promise anything, but current plan is that I should start some work early next year.

**Comment by brainfoolong** [comment]
Hi @piotrtomiak - It has been a long time since we last talked here. Was time to do anything regarding to this problem? I know, every power goes in integrating useless AI things everywhere, but maybe there is a window for that?

We still juggling with unnecesarry package.json all over the place, just for web types.
